core/coreapi/unixfs.go

```diff
@@ -254,3 +259,29 @@ func (api *UnixfsAPI) lsFromLinks(ctx context.Context, ndlinks []*ipld.Link, set
 func (api *UnixfsAPI) core() *CoreAPI {
 	return (*CoreAPI)(api)
 }
+
+func (api *UnixfsAPI) AppendMetadata(metaMap map[string]interface{}, opts ...options.UnixfsAddOption) error {
+	if metaMap == nil {
+		return nil
+	}
+
+	settings, _, err := options.UnixfsAddOptions(opts...)
+	if err != nil {
+		return err
+	}
+
+	b, err := json.Marshal(metaMap)
+	if err != nil {
+		return err
+	}
+
+	buf := bytes.Buffer{}
+	if settings.TokenMetadata != "" {
+		buf.WriteString(settings.TokenMetadata)
+	}
+	buf.WriteString(string(b))
```

Comment:
Are you sure they can be merged this way? It seems that you are concating two maps without merging the {} first

Reply:
settings.TokenMetadata is a string.